src/resources/2021/CrystalBall.ts

```diff
@@ -1,38 +1,29 @@
 import {
   Location,
   Monster,
-  myTurncount,
   toLocation,
   toMonster,
+  availableAmount,
+  Item,
+  visitUrl,
 } from "kolmafia";
 import { get } from "../../property";
 
+export const orb = Item.get("miniature crystal ball");
+export function have(): boolean {
+  return availableAmount(orb) > 0;
+}
 const parsedProp = () =>
   get("crystalBallPredictions")
     .split("|")
     .map((element) => element.split(":") as [string, string, string])
     .map(
-      ([turncount, location, monster]) =>
-        [parseInt(turncount), toLocation(location), toMonster(monster)] as [
-          number,
-          Location,
-          Monster
-        ]
+      ([, location, monster]) =>
+        [toLocation(location), toMonster(monster)] as [Location, Monster]
     );
 
-export function currentPredictions(withFree = true): Map<Location, Monster> {
-  const predictions = parsedProp();
-  const freeCondition = (predictedTurns: number, turns: number) =>
-    predictedTurns === turns;
-  const nonFreeCondition = (predictedTurns: number, turns: number) =>
-    predictedTurns + 1 === turns;
-  return new Map(
-    predictions
-      .filter(
-        ([turncount]) =>
-          nonFreeCondition(turncount, myTurncount()) ||
-          (withFree && freeCondition(turncount, myTurncount()))
-      )
-      .map(([, location, monster]) => [location, monster])
-  );
+export function ponder(): Map<Location, Monster> {
+  if (!have()) return new Map();
+  visitUrl("inventory.php?action=ponder", false);
```

Comment:
also i am still going to approve this because i think it's fine but i just want to register this here in the PR -- are we -positive- there is no real way libram will be calling this in the middle of some sort of chained NC/combat string, like the LOV tunnel? i expect not! you probably would need to put it in a postadventure script. but i am curious exactly -how- it would fuck up if you did that, and if there is room for an error catch here that simply returns the most recent predictions if a user messed up like that somehow.

Reply:
_libram_ won't be calling this in the middle of some chained NC/combat string, but a player could. My general attitude is that if the product we produce is a drain cleaner, we aren't responsible for people who use it as an enema.

Reply:
Digging around, I have a way to check to see if you're in a multiFight, or if a choice follows fight, or whatever, but I don't see a way to determine if you're currently in combat

Reply:
:eyes: